src/main.rs

```diff
@@ -53,176 +53,130 @@ fn buffer_repaint(stdout: &mut Stdout, buffer: &LineBuffer, prompt_offset: u16)
     Ok(())
 }
 
+// this fn is totally ripped off from crossterm's examples
+// it's really a diagnostic routine to see if crossterm is
+// even seeing the events. if you press a key and no events
+// are printed, it's a good chance your terminal is eating
+// those events.
+fn print_events(stdout: &mut Stdout) -> Result<()> {
+    loop {
+        // Wait up to 5s for another event
+        if poll(Duration::from_millis(5_000))? {
+            // It's guaranteed that read() wont block if `poll` returns `Ok(true)`
+            let event = read()?;
+
+            // just reuse the print_message fn to show events
+            print_message(stdout, &format!("Event::{:?}", event))?;
+
+            // hit the esc key to git out
+            if event == Event::Key(KeyCode::Esc.into()) {
+                break;
+            }
+        } else {
+            // Timeout expired, no event for 5s
+            print_message(stdout, "Waiting for you to type...")?;
+        }
+    }
+
+    Ok(())
+}
+
 fn main() -> Result<()> {
     let mut stdout = stdout();
 
     terminal::enable_raw_mode()?;
+    // quick command like parameter handling
+    let args: Vec<String> = std::env::args().collect();
+    // if -k is passed, show the events
+    if args.len() > 1 && args[1] == "-k" {
+        print_message(&mut stdout, "Ready to print events:")?;
+        print_events(&mut stdout)?;
+    } else {
```

Comment:
One suggestion I think would help readability:

If `-k` just does a "print event loop", could we just return after the `print_events` above. This would keep you from having to use `else` and indent everything under it.

Reply:
ok. I added an early return instead of else.